src/xenia/kernel/xam/xam_net.cc

```diff
@@ -1227,47 +1249,163 @@ dword_result_t XampXAuthStartup_entry(pointer_t<XAUTH_SETTINGS> setttings) {
+
+  return hInternet->handle();
+}
+DECLARE_XAM_EXPORT1(NetDll_XHttpOpen, kNetworking, kImplemented);
+
+dword_result_t NetDll_XHttpCrackUrl_entry(
+    dword_t caller, lpstring_t pcszUrl, dword_t dwUrlLength, dword_t flags,
+    pointer_t<URL_COMPONENTS> pUrlComponents) {
+  XELOGI("HttpCrackUrl {} {} {}", pcszUrl, dwUrlLength, flags);
+
+  if (dwUrlLength < 1) {
+    return FALSE;
+  }
+  return TRUE;
+}
+DECLARE_XAM_EXPORT1(NetDll_XHttpCrackUrl, kNetworking, kStub);
+
+dword_result_t NetDll_XHttpConnect_entry(dword_t caller, dword_t hSession,
+                                         lpstring_t host, dword_t port,
+                                         dword_t flags) {
+  std::string hostname = "127.0.0.1";
+  uint16_t connectPort = uint16_t(port);
+  if (hostname.length() < 1) {
+    hostname = "127.0.0.1";
+  }
+  if (connectPort == 0) {
+    connectPort = 80;
+  }
+  XELOGI("HttpConnect: {} {} {} {}", hSession, hostname, connectPort, flags);
+
+  if (!hInternet) {
+    hInternet = new HINTERNET(kernel_state());
+    hInternet->CreateSessionHandle("Xenia");
+  }
+
+  X_HANDLE hConnect = hInternet->CreateConnectionHandle(hostname, port);
+  return hConnect;
+}
+DECLARE_XAM_EXPORT1(NetDll_XHttpConnect, kNetworking, kImplemented);
+
+dword_result_t NetDll_XHttpOpenRequest_entry(dword_t caller, dword_t hConnect,
+                                             lpstring_t verb, lpstring_t path,
+                                             lpstring_t version,
+                                             lpstring_t referrer,
+                                             lpstring_t reserved,
+                                             dword_t flag) {
+  std::string url = path ? (std::string)path : "";
+  std::string method = verb ? (std::string)verb : "";
+  XELOGI("HttpOpenRequest: {} {} {}", hConnect, method, url);
+  if (hConnect <= 1) {
+    return 1;
+  }
+  if (!hInternet) {
+    return 1;
+  }
+  X_HANDLE hRequest = hInternet->CreateRequestHandle(url, method);
+  return hRequest;
+}
+DECLARE_XAM_EXPORT1(NetDll_XHttpOpenRequest, kNetworking, kImplemented);
+
+dword_result_t NetDll_XHttpCloseHandle_entry(dword_t caller, dword_t handle) {
+  XELOGI("HttpCloseHandle: {}", handle);
+  if (handle == 1) {
+    return FALSE;
+  }
+  if (!hInternet) {
+    return FALSE;
+  }
+  hInternet->ReleaseHandle();
+  hInternet = nullptr;
+  return TRUE;
+}
+DECLARE_XAM_EXPORT1(NetDll_XHttpCloseHandle, kNetworking, kImplemented);
 
 dword_result_t NetDll_XHttpSetStatusCallback_entry(dword_t caller,
-                                                   dword_t handle,
-                                                   lpdword_t callback_ptr,
+                                                   dword_t hInternet,
+                                                   lpvoid_t callback_ptr,
                                                    dword_t flags, dword_t unk) {
-  return 1;
+  XELOGI("HttpSetStatusCallback: {}", hInternet);
+  return NULL;
```

Comment:
Do not return null from calls that return dword_result_t, It's better to return 0 or even better some X_STATUS code